packages/image_picker/image_picker/example/ios/RunnerTests/PickerSaveImageToPathOperationTests.m

```diff
@@ -0,0 +1,85 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <OCMock/OCMock.h>
+#import <UniformTypeIdentifiers/UniformTypeIdentifiers.h>
+#import <XCTest/XCTest.h>
+
+@import image_picker;
+
+@interface PickerSaveImageToPathOperationTests : XCTestCase
+
+@end
+
+@implementation PickerSaveImageToPathOperationTests
+
+- (void)testSaveWebPImage API_AVAILABLE(ios(14)) {
+  NSURL *imageURL = [[NSBundle bundleForClass:[self class]] URLForResource:@"webpImage"
+                                                             withExtension:@"webp"];
+  NSItemProvider *itemProvider = [[NSItemProvider alloc] initWithContentsOfURL:imageURL];
+  PHPickerResult *result = [self createPickerResult:itemProvider
+                                     withIdentifier:UTTypeWebP.identifier];
+
+  [self saveImageToPath:result];
+}
+
+- (void)testSavePNGImage API_AVAILABLE(ios(14)) {
+  NSURL *imageURL = [[NSBundle bundleForClass:[self class]] URLForResource:@"pngImage"
+                                                             withExtension:@"png"];
+  NSItemProvider *itemProvider = [[NSItemProvider alloc] initWithContentsOfURL:imageURL];
+  PHPickerResult *result = [self createPickerResult:itemProvider
+                                     withIdentifier:UTTypeWebP.identifier];
+
+  [self saveImageToPath:result];
+}
+
+- (void)testSaveJPGImage API_AVAILABLE(ios(14)) {
+  NSURL *imageURL = [[NSBundle bundleForClass:[self class]] URLForResource:@"jpgImage"
+                                                             withExtension:@"jpg"];
+  NSItemProvider *itemProvider = [[NSItemProvider alloc] initWithContentsOfURL:imageURL];
+  PHPickerResult *result = [self createPickerResult:itemProvider
+                                     withIdentifier:UTTypeWebP.identifier];
+
+  [self saveImageToPath:result];
+}
+
+- (void)testSaveGIFImage API_AVAILABLE(ios(14)) {
+  NSURL *imageURL = [[NSBundle bundleForClass:[self class]] URLForResource:@"gifImage"
+                                                             withExtension:@"gif"];
+  NSItemProvider *itemProvider = [[NSItemProvider alloc] initWithContentsOfURL:imageURL];
+  PHPickerResult *result = [self createPickerResult:itemProvider
+                                     withIdentifier:UTTypeWebP.identifier];
+
+  [self saveImageToPath:result];
+}
+
+- (PHPickerResult *)createPickerResult:(NSItemProvider *)itemProvider
```

Comment:
In ObjC naming, the thing right before the argument should name the argument; the argument isn't a result here. This should be called something like `createPickerResultWithProvider:identifier:`

Reply:
I completely forgot this rule. I have switched to swift for a long time ago and didn't write any objective-c code :D